spec/unit/ssl/state_machine_spec.rb

```diff
@@ -318,6 +344,32 @@ def expect_lockfile_to_contain(pid)
 
       expect(File).to_not exist(Puppet[:localcacert])
     end
+
+    context 'when verifying CA cert bundle' do
+      before :each do
+        allow(cert_provider).to receive(:load_cacerts).and_return(nil)
+        stub_request(:get, %r{puppet-ca/v1/certificate/ca}).to_return(status: 200, body: cacert_pem)
+        allow(cert_provider).to receive(:save_cacerts)
+      end
+
+      it 'verifies CA cert bundle if a ca_fingerprint is given case-insensitively' do
+        Puppet[:log_level] = :info
+        machine = described_class.new(digest: 'SHA256', ca_fingerprint: 'caacf69bbbcdad9dbcda92dd2da3608b639d1aea4c314d6cc6823cdb32d8e0f8')
```

Comment:
Are they always only ever going to be SHA256? Is it assumed that if a SHA256 works, so would a SHA512, or any other algorithm given, or should those also be being tested here?

Reply:
It'd be good to verify if a non-default digest is passed, that it passes as well. I'll add a test for that.

Note it's not currently possible to pass the digest from the command line,

Reply:
Added test in https://github.com/puppetlabs/puppet/compare/44241e0b41ef6c727ad462366f20e95e742f0acb..43472eef86e440ee3119f8d5c041bf2fde3e329e